source/sundials/kinsol.cc

```diff
@@ -228,7 +228,7 @@ namespace SUNDIALS
 
     // helper function to create N_Vectors compatible with different versions
     // of SUNDIALS
-    const auto make_compatible_nvector_view = [this](auto &v) {
+    const auto make_compatible_nvector_view = [](auto &v) {
       return internal::make_nvector_view(v
 #  if !DEAL_II_SUNDIALS_VERSION_LT(6, 0, 0)
                                          ,
```

Comment:
Does this work with both branches of `#  if !DEAL_II_SUNDIALS_VERSION_LT(6, 0, 0)`? What with the variable `kinsol_ctx` that is needed there?

Reply:
You are of course correct. Fixed.